cfme/common/provider.py

```diff
@@ -50,6 +52,7 @@ class BaseProvider(Taggable, Updateable, SummaryMixin):
     _properties_region = None
     add_provider_button = None
     save_button = None
+    appliance = get_or_create_current_appliance()
```

Comment:
Noted, and I'll make this update.

FWIW, the function name seems to imply that it will return an existing current appliance or create a new one if none exists. Description of the call on class vs its instance as the driver for usage is helpful.